### DISCUSSION TITLE
Mocking ScenarioContext

### DISCUSSION THREAD

**Discussion by phatcher** [reqnroll/Reqnroll#222] [Ideas]
I have fair number of extensions methods but I can't write unit tests, only integration tests, since `ScenarioContext` is not fully mockable from `IScenarioContext`.

It might be as simple as making `IScenarioContext` inherit from `IDictionary<string, object>`

Any objects to a PR along these lines or a different approach?

**Comment by Code-Grump** [comment]
What aspect of the context do you want to mock, but can't?

**Reply by phatcher** [reply]
Getting/Settting data in the context e.g. `context[name]`, I'm storing the responses from API calls/controllers under standard names so I can write common patterns, for example storing the status code of a web call.

**Reply by Code-Grump** [reply]
Whilst `ScenarioContext` is great for getting started, have you considered defining your own context classes to handle your needs?

https://docs.reqnroll.net/latest/automation/context-injection.html

**Reply by phatcher** [reply]
The problem with specific contexts, is that it ties implementers to using them, whereas with the dictionary you can write generic extensions that are easily adoptable

**Comment by gasparnagy** [comment]
Besides the suggestion of @Code-Grump about context injection, for code consistency, I think adding `IDictionary<string, object>` to the `IReqnrollContext` (the base interface of all `I*Context`) interface would be useful, so a PR is welcome.

**Reply by phatcher** [reply]
I'll have a go at producing one
